### PR TITLE
#1365: DatePicker // update react-dates and expose more css variables (closes #1365)

### LINKED ISSUE #1365: DatePicker // update react-dates and expose more css variables
## Description
The new version of `react-dates` expose additional props that we missed in the switch from `rc-datepicker` to the new datepicker (eg. "show on top" flag).
Update the `react-dates` version and expose the props that could be helpful.

Moreover, some css style variable is missing (like the calendar background color). Expose them

### PR DESCRIPTION
**Pull request by veej**
Closes #1365

### CODE REVIEW

**Comment by alessandravilla** [comment]
tutto ok, unica cosa il placeholder dovrebbe essere #9098a7

File: src/DatePicker/DatePicker.tsx [hunk 1]
```diff
@@ -109,37 +98,31 @@ const calendarIcon = (
...
  */
-@props(Props)
 export class DatePicker extends React.PureComponent<DatePicker.Props, State> {
+  dayRenderer: (day: moment.Moment) => JSX.Element;
+
   componentWillMount() {
     if (this.props.locale) {
       moment.locale(this.props.locale);
     }
+    this.dayRenderer = this.dayRendererFactory();
```
Comment (giogonzo) [inline src/DatePicker/DatePicker.tsx]: It's not 100% clear why the indirection `dayRenderer`/`dayRendererFactory` is needed. Maybe we can discuss this offline a few minutes?

File: src/DatePicker/datePicker.scss [hunk 2]
```diff
@@ -1,224 +1,246 @@
...
+$selected-day-color: $brc-darkGreyBlue !default;
+$selected-day-hover-background: $brc-azure !default;
+$selected-day-hover-color: $brc-darkGreyBlue !default;
+$hover-interval-day-background: $brc-azure10 !default;
+$hover-interval-day-border: $brc-azure !default;
+$hover-interval-day-color: $brc-darkGreyBlue !default;
+$selected-interval-day-background: $brc-azure30 !default;
+$selected-interval-day-border: $brc-azure10 !default;
+$selected-interval-day-color: $brc-darkGreyBlue !default;
+$outside-day-background: $brc-white !default;
```
Comment (giogonzo) [inline src/DatePicker/datePicker.scss]: as discussed offline, I would normalize the naming of variables to have the `-hover` modifier always at the end + adding a prefix/namespace for the single component (`date-picker-<variable-name>`)

File: src/DatePicker/DatePicker.tsx (hunk 1 above)
Comment (giogonzo) [reply src/DatePicker/DatePicker.tsx]: Discussed offline and now clearer. Just for the sake of completeness, I would add a link to the react-dates issue explaining the need and rationale behind this @veej

**Pull request merged**